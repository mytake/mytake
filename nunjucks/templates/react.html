{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
