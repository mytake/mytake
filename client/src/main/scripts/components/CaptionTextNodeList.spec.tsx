import * as React from "react";
import * as renderer from "react-test-renderer";
import CaptionTextNodeList from "./CaptionTextNodeList";
import { videoFactFast, videoNodes } from "../utils/testUtils";

const onMouseUp = jest.fn();
const onScroll = jest.fn();

const eventHandlers = {
  onMouseUp: onMouseUp,
  onScroll: onScroll
};

jest.mock("./CaptionTextNode", () => ({
  default: "CaptionTextNode"
}));

test("CaptionTextNodeList", () => {
  const tree = renderer
    .create(
      <CaptionTextNodeList
        documentNodes={videoNodes}
        eventHandlers={eventHandlers}
        videoFact={videoFactFast}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
