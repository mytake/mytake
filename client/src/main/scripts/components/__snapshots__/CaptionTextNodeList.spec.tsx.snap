// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptionTextNodeList 1`] = `
<div
  className="document__text document__text--caption"
  onMouseUp={[Function]}
  onScroll={[Function]}
>
  <CaptionTextNode
    documentNode="Good evening I'm Martha Raddatz from ABC News."
    speaker="Raddatz"
  />
  <CaptionTextNode
    documentNode="And I'm Anderson Cooper from CNN. We want to welcome you to Washington University in St. Louis for the second presidential debate between Hillary Clinton and Donald Trump sponsored by the Commission on Presidential Debates. tonight's debate is a Town Hall format, which gives voters a chance to directly ask the candidates questions. Martha and I will ask follow-up questions but the night really belongs to the people in this room and to people across the country who have submitted questions online."
    speaker="Cooper"
  />
</div>
`;
